pkg/image/controller/trigger/trigger_controller.go

```diff
@@ -0,0 +1,399 @@
+package trigger
+
+import (
+	"fmt"
+	"math"
+	"strconv"
+	"strings"
+	"time"
+
+	"github.com/golang/glog"
+
+	"k8s.io/kubernetes/pkg/api"
+	"k8s.io/kubernetes/pkg/api/errors"
+	"k8s.io/kubernetes/pkg/api/unversioned"
+	"k8s.io/kubernetes/pkg/client/cache"
+	kcoreclient "k8s.io/kubernetes/pkg/client/clientset_generated/internalclientset/typed/core/internalversion"
+	"k8s.io/kubernetes/pkg/client/record"
+	"k8s.io/kubernetes/pkg/controller"
+	utilruntime "k8s.io/kubernetes/pkg/util/runtime"
+	"k8s.io/kubernetes/pkg/util/wait"
+	"k8s.io/kubernetes/pkg/util/workqueue"
+
+	imageapi "github.com/openshift/origin/pkg/image/api"
+	"github.com/openshift/origin/pkg/image/trigger"
+)
+
+const (
+	// maxRetries is the number of times an image stream will be retried before it is dropped out of the queue.
+	maxRetries          = 5
```

Comment:
Although based on your use of defaultResourceFailureDelay, it seems that you simply cap to 30s, so the comment for maxRetries does not hold true (the image stream will never be dropped).